src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -237,11 +230,14 @@ public Conference(Videobridge videobridge,
         this.includeInStatistics = enableLogging;
         this.conferenceName = conferenceName;
 
-        speechActivity = new ConferenceSpeechActivity(this);
+        speechActivity = new ConferenceSpeechActivity(new SpeechActivityListener());
         updateLastNEndpointsFuture = TaskPools.SCHEDULED_POOL.scheduleAtFixedRate(() -> {
             try
             {
-                lastNEndpoints.update();
+                if (speechActivity.updateLastNEndpoints())
+                {
+                    lastNEndpointsChanged();
```

Comment:
Why not have this be an event defined in `ConferenceSpeechActivity.Listener` ?

So here we'd just do `speechActivity.updateLastNEndpoints()` and, if something changed, the event would be fired (instead of return a bool?)

Reply:
I want updating the bitrate controllers to run in a separate task (it may send endpoint messages), but avoid creating the task if the list hasn't changed. I see now that I have some cases that lead to unnecessary tasks, so I'm making some changes.